clients/spark-extensions-base/src/test/java/org/projectnessie/spark/extensions/AbstractSparkSqlTest.java

```diff
@@ -299,6 +299,48 @@ void testCreateTagInFrom() throws NessieNotFoundException {
         .hasMessage("Named reference 'testBranch' not found");
   }
 
+  @Test
+  void testCreateReferenceFromHashOnNonDefaultBranch() throws NessieNotFoundException {
+    assertThat(sql("CREATE BRANCH %s IN nessie FROM main", refName))
+        .containsExactly(row("Branch", refName, hash));
+    assertThat(api.getReference().refName(refName).get()).isEqualTo(Branch.of(refName, hash));
+    sql("USE REFERENCE %s IN nessie", refName);
+    sql("CREATE TABLE nessie.db.tbl (id int, name string)");
+    sql("INSERT INTO nessie.db.tbl select 23, \"test\"");
+    String newHash =
+        api.getCommitLog()
+            .refName(refName)
+            .maxRecords(1)
+            .get()
+            .getLogEntries()
+            .get(0)
+            .getCommitMeta()
+            .getHash();
+
+    String tempRef = refName + "_temp";
+    sql("CREATE BRANCH %s IN nessie FROM %s", tempRef, refName);
+    assertThat(api.getReference().refName(tempRef).get()).isEqualTo(Branch.of(tempRef, newHash));
+
+    String tag = refName + "_temp_tag";
+    sql("CREATE TAG %s IN nessie FROM %s", tag, refName);
+    assertThat(api.getReference().refName(tag).get()).isEqualTo(Tag.of(tag, newHash));
+
+    assertThat(sql("DROP BRANCH %s IN nessie", refName)).containsExactly(row("OK"));
```

Comment:
those 3 DROP statements aren't really needed as they are being cleaned up automatically

Reply:
other testcases also has the same pattern. So, I followed the same.

Reply:
yes I know, and this should rather be cleaned up. We should just have separate tests for dropping a branch/tag and not pollute existing tests with drop statements

Reply:
testCreateBranch, testCreateTag, testCreateBranchIn, testCreateTagIn, testCreateBranchInFrom, testCreateTagInFrom
all these has it, should I remove from them aswell or let's keep it?